atomix/cluster/src/test/java/io/atomix/raft/RaftFailOverTest.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright © 2020 camunda services GmbH (info@camunda.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.atomix.raft;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import io.atomix.storage.journal.Indexed;
+import java.util.List;
+import java.util.Map;
+import org.junit.Ignore;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameter;
+import org.junit.runners.Parameterized.Parameters;
+
+@RunWith(Parameterized.class)
+public class RaftFailOverTest {
+
+  @Rule @Parameter public RaftRule raftRule;
+
+  @Parameters(name = "{index}: {0}")
+  public static Object[][] reprocessingTriggers() {
+    return new Object[][] {
+      new Object[] {RaftRule.withBootstrappedNodes(3)},
+      new Object[] {RaftRule.withBootstrappedNodes(4)},
+      new Object[] {RaftRule.withBootstrappedNodes(5)}
+    };
+  }
+
+  @Test
+  public void shouldCommitEntriesAfterFollowerShutdown() throws Throwable {
+    // given
+    final var entryCount = 20;
+    raftRule.appendEntries(entryCount);
+    raftRule.awaitCommit(entryCount);
+    raftRule.shutdownFollower();
+
+    // when
+    raftRule.appendEntries(entryCount);
+
+    // then
+    // 40 zeebe entries and 1 initial entry
```

Comment:
We can get the index from `appendEntry` and use that to await commit, instead of estimating the commit index.